pkg/chartutil/load.go

```diff
@@ -214,6 +215,68 @@ func LoadFile(name string) (*chart.Chart, error) {
 	return LoadArchive(raw)
 }
 
+// Getting Helm client home directory.
+
+func getHelmHomeDir() string {
+	var home string
+	if home = os.Getenv("$HELM_HOME"); home != "" {
+		return home + string(filepath.Separator)
+	}
+	if runtime.GOOS == "windows" {
+		home = os.Getenv("HOMEDRIVE") + os.Getenv("HOMEPATH")
+		if home == "" {
+			home = os.Getenv("USERPROFILE")
+		}
+	} else {
+		home = os.Getenv("HOME")
+	}
+	home += string(filepath.Separator) + ".helm" + string(filepath.Separator)
+
+	return home
+}
+
+// LoadIgnoreFiles loads global helmignore file from the default helm client
+// folder and if exists, helmignore file from the current folder into
+// a buffer and return a reader.
+func LoadIgnoreFiles(dir string) (*bytes.Reader, error) {
+	buff := bytes.NewBuffer(nil)
+
+	topdir, err := filepath.Abs(dir)
+	if err != nil {
+		return nil, err
+	}
+	topdir += string(filepath.Separator)
+
+	f, err := os.Open(topdir + ignore.HelmIgnore)
+	if err != nil {
+		if !os.IsNotExist(err) {
+			return nil, err
+		}
+	} else {
+		defer f.Close()
+		_, err = io.Copy(buff, f)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	homeDir := getHelmHomeDir()
```

Comment:
So there's an unwritten rule in the Helm codebase that nothing inside of `pkg` can assume knowledge of the $HELM_HOME. This is so that other programs can be written using the `pkg` libraries.

When I was talking about this with @adamreese, he had an excellent idea:

- Change `LoadDir()`'s function signature to be `LoadDir(dir string, ignoreFiles ...string) (*chart.Chart, error)`
- Change `Load()`'s signature the same way

Then you can pass in the path to the ignore file whenever you can calculate it. And since we use the `...` notation, that parameter can be omitted. So `LoadDir("foo")` will still work. Then we also don't break backward compatibility.

This will mean fixing a few of the cmd/helm files to call Load or LoadDir with the right args, but @adamreese and I can help track those down. It might even be easier to do that in subsequent pull requests instead of doing it all in one.